ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/rewrite/testtool/infrastructure/LedgerTestContext.scala

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.rewrite.testtool.infrastructure
+
+import java.time.Instant
+
+import com.digitalasset.daml.lf.data.Ref
+import com.digitalasset.daml.lf.language.Ast
+import com.digitalasset.ledger.api.v1.ledger_offset.LedgerOffset
+import com.digitalasset.ledger.api.v1.transaction.Transaction
+import com.digitalasset.ledger.api.v1.value.{Identifier, Value}
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{ExecutionContext, Future}
+
+final class LedgerTestContext(val applicationId: String, bindings: LedgerBindings)(
+    implicit val ec: ExecutionContext)
+    extends ExecutionContext {
+
+  override def execute(runnable: Runnable): Unit = ec.execute(runnable)
+  override def reportFailure(cause: Throwable): Unit = ec.reportFailure(cause)
+
+  val ledgerId: Future[String] = bindings.ledgerId
+
+  val offsetAtStart: Future[LedgerOffset] = bindings.ledgerEnd
```

Comment:
I think this probably should be passed in as a constructor parameter as a non-`Future` value. This has the smell of potential flakiness.

Reply:
I see your point, I'll investigate the point further.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/2282/commits/e41532036369394d18a9a5278323f476d24fce77

Thanks for the tip given offline.